### ISSUE TITLE
IKEA RODRET (E2201) keeps dropping off the network

### ISSUE THREAD

**Issue by aatifk** [Koenkk/zigbee2mqtt#20758]
### What happened?

After the RODRET switch connected successfully, it worked fine for about an hour and then it dropped off the network automatically. I force removed the device and added it again, and same issue happened again after about an hour. It won't maintain a connection to the network.

Worth noting that in Z2M devices, it still shows that it's online. However, no commands go through, and there's no connection to it in the Network Map either. 

### What did you expect to happen?

The switch should have stayed connected.

### Zigbee2MQTT version

1.35.1

### Adapter firmware version

20230507

### Adapter

SONOFF Zigbee Dongle P

### Setup

Native Linux Install

**Comment by danieledwardgeorgehitchcock** [comment]
Hi,

What sort of LQI figures are you getting when the device is connected and functional?

Also, what does your network map say the route is back to the coordinator?

Is the device a fair distance away from the coordinator / any Zigbee routing devices?

**Comment by aatifk** [comment]
I've been getting an LQI of 130+ when it's functional, and my house is covered in routers. I'd say it was about 10 feet from the coordinator though when I was trying this earlier. 

As an update, I force-paired it to a nearby router (permit join on the router only) and it's stayed connected for the past 24 hours. Will see how it holds up in the longer run.

**Comment by aatifk** [comment]
Update: It kept on for about 3 days and dropped off again. The router is hardly 5 ft from the device and yet it dropped again. Any thoughts?

**Comment by Taxxian** [comment]
I had a probably connected problem;

I could not even pair the Rodret to the Sonoff P stick, it dropped of the network while pairing.

Solution: I skipped the  Coordinator and paired it to a Sonoff Switch acting as a hub, and since this hub is always closer to the Rodret than the Coordinator it now works fine.

**Comment by MattWestb** [comment]
The problem is some routers is not good ones and all IKEA controllers is using Zigbee 3 security (on current production firmware) and if the parent is not doing it OK the device think its OK but the router have timing it out and is deleting it in its children table so the network cant finding it.

**Comment by RubenKelevra** [comment]
I bought two today to check them out, but can't connect them to my network.

Pressing the link button for >10 seconds does nothing, not even lighting up the LED on them.

I've checked the batteries I put in them, they are happily at around 1.5V.

Is there any trick or something I'm missing? 🤔

**Comment by MattWestb** [comment]
4 pressing the paring bottom is resetting the controller and going in paring mode.
Long press is touch light paring mode that being used for direct paring with light without coordinator.

**Comment by RubenKelevra** [comment]
@MattWestb thanks for the fast help!

Will report back if I see the same issue with my devices

**Comment by RubenKelevra** [comment]
@MattWestb I can't confirm the issue you're having with the device. I got two of them and use one of them daily and one maybe biweekly to turn lights on/off. They both work fine for 3 month now with no outage or delay after pressing.

**Comment by wwerther** [comment]
Hi,
I tried today and I'm also having issues with these devices. I have two of them and pairing works fine. After that no commands are issued and send when I press a button.
They work fine when I directly peer them to a TRETAKT smart plug. But when I try to peer with my coordinator no messages are transferred. 
My coordinator is a cod-m device connected via network. All other devices in my zigbee net work fine.

**Comment by RubenKelevra** [comment]
@wwerther do you use fw 1.0.57?

**Comment by wwerther** [comment]
> @wwerther do you use fw 1.0.57?
As I am not sure which FW version we're exactly talking about, I will just list all that I can get:

Coordinator: 	
CZC v1.0, ESP-FW: 1.0.0 (codm-v1-prod), 
CC2652P hw version:	CC2652P7
CC2652P fw rev:	20230507

Home-Assistant: 2024.7.4
addon Zigbee2Mqtt: Current version: 1.39.1-1
Coordinator-Type zStack3x0
Coordinator-Version 20230507
Frontend-Version 0.7.4
Zigbee Herdsman Konverter Version 19.72.0
Zigbee Herdsman Version 0.55.3

Rodret-Version: 1.0.47

interestingly I get "online" messages. So today I can see that coordinator sees device was last seen 3hours, 4hours ago. anyway Keypress is not recognized (no event visible in logs)

 please let me know if I can support in creating debug logs etc. 

Thanks

**Comment by Minims** [comment]
Hello,

I have the same issue since 2 week. 
It has worked great for 1 year. 

To make it work again i need to put a new battery, but it works only 3 days max.
The battery is not drained, still 1.48v on the battery.

The device keeps lopping on online message

**Comment by RubenKelevra** [comment]
> Rodret-Version: 1.0.47

So firmware 1.0.57 isn't shown as update for this device?

**Comment by RubenKelevra** [comment]
> This issue is stale because it has been open 180 days with no activity. Remove stale label or comment or this will be closed in 30 days

@Koenkk the bot is malfunctioning. It hasn't been 180 since the last activity here

**Comment by Koenkk** [comment]
@RubenKelevra the comment text was wrong, I'll fix it, thanks for letting me know!

**Comment by tobiasbayer** [comment]
Was this issue closed accidentially? I have the same issue and I cannot find a solution or a reason for the observed behavior in this thread.

**Comment by RubenKelevra** [comment]
@tobiasbayer nah, just the worst anti feature on Github: A bot which automatically closes valid bug reports if you don't constantly ping these as users m(

**Comment by tobiasbayer** [comment]
Maybe @Koenkk could reopen it?

**Comment by RubenKelevra** [comment]
@tobiasbayer yeah only team members can do that. That's why this bot is wasting so much time. It's a constant bumping and pinging just to keep valid bug reports open all over Github. It's super annoying, and that basically to fudge the numbers of open bug reports.

**Comment by devclef** [comment]
Hi, 

Just sharing the comment here:
https://github.com/Koenkk/zigbee2mqtt/discussions/23243#discussioncomment-12740348

Was useful for me. I had a few new rodret and styrbar devices I picked up recently during Ikea's sell out. I'd either get them to join, but not report anything or they'd connect/disconnect. I tried various batteries, multiple resets, slow pressing buttons, not pressing buttons, pressing buttons reasonably fast, etc. Nothing worked.

Ultimately I tried the comment above, and pressed buttons like an absolute lunatic until I saw successful configured in the logs and then for another 10-30 seconds for good measure (not saying required, just what I did). I successfully did this on 3 additional devices 'first try', in addition to the two devices I had been struggling with. 

They have stayed joined, actions showing as expected and able to create automations within Home Assistant with them. Jokes aside, I was clicking as fast as I could all the buttons during the joining process.

**Comment by RubenKelevra** [comment]
@devclef well non force removal requires the device to "listen" but those battery operated devices go to sleep basically right after pressing a button again. So if that's where you were struggling that's easy to explain.

But the pairing should work fine, as it should stay awake until pairing is completed. However there's also a "link to nearby light bulbs" function in those devices. This can sometimes cause issues when you're anywhere near a light bulb and some light bulbs are more sensitive to that than others. If you wanna avoid it cut the power to the light bulbs in the room where your hub is and try again to pair after forced removal of the device from z2m.

**Comment by devclef** [comment]
> [@devclef](https://github.com/devclef) well non force removal requires the device to "listen" but those battery operated devices go to sleep basically right after pressing a button again. So if that's where you were struggling that's easy to explain.
> 
> But the pairing should work fine, as it should stay awake until pairing is completed. However there's also a "link to nearby light bulbs" function in those devices. This can sometimes cause issues when you're anywhere near a light bulb and some light bulbs are more sensitive to that than others. If you wanna avoid it cut the power to the light bulbs in the room where your hub is and try again to pair after forced removal of the device from z2m.

I did also have an issue with removing them during some of the attempts, other attempts I could remove without force.

I thought the linking was from a long press on the pair button, compared to 4x to trigger zigbee pairing?

I did these test from sitting in my office chair, there is an ikea bulb in the ceiling a couple meters away. Perhaps this was causing a problem though i'd of assumed I'd need to be quite close for such binding. 

The device would finish interviewing, configuring successful, then disappear seconds later. Or stay joined but not provide any data - null on battery, action, etc. 

It is quite reproducible for me with these battery powered ikea devices, and consistently able to pair them with more frequent pushing of the buttons during pairing. I do not recall this being required with earlier devices, year(s) ago with zigbee2mqtt. 

On the devices which stayed joined but non functional, there was only one cluster bound to the coordinator. Checking a 'working device' now, I see 4 clusters bound. 

I have more devices so happy to create this with debug and such if desired, I honestly was going to put this in 'battery devices can be finnicky sometimes bucket' and carry on :) Was just going to share for others as something to try.

**Comment by RubenKelevra** [comment]
> I thought the linking was from a long press on the pair button, compared to 4x to trigger zigbee pairing?

Yeah except I got other remotes (non IKEA) and they have also two different pair modes, but still bond sometimes randomly to a light nearby if I try to put them into the Zigbee network.

That's why I pointed this out. Not because I found this was happening with the 5 IKEA remotes I got.

**Comment by andrei-lazarov** [comment]
Hi. I didn't follow the thread, but regarding pairing IKEA remotes: 
**The router through which you pair the remotes is very important.** 
Using rechargeable batteries, USB 2.0 extension cable for the coordinator, reducing interference might also help.

I can consistently fail pairing my STYRBAR (shows success, but Actions not reported) through a Tuya Telink bulb...
Power-off the bulb and pairing works immediately, 100% successful, all Actions reported.
Sometimes clicking the drop-down on Permit join and selecting coordinator is enough.

I consider Telink devices (IEEE adress starts with `0xa4c138`) bad routers. Usually Tuya and Tuya re-brands, AwoX
